internal/cli/email_templates.go

```diff
@@ -0,0 +1,292 @@
+				var err error
+				current, err = cli.api.EmailTemplate.Read(apiEmailTemplateFor(inputs.Template))
+				return err
+			})
+			if err != nil {
+				return fmt.Errorf("Unable to get the email template '%s': %w", inputs.Template, err)
+			}
+
+			if err := emailTemplateFrom.AskU(cmd, &inputs.From, current.From); err != nil {
+				return err
+			}
+
+			if err := emailTemplateSubject.AskU(cmd, &inputs.Subject, current.Subject); err != nil {
+				return err
+			}
+
+			// TODO(cyx): we can re-think this once we have
+			// `--stdin` based commands. For now we don't have
+			// those yet, so keeping this simple.
+			if err := emailTemplateBody.EditorPromptU(
+				cmd,
+				&inputs.Body,
+				current.GetBody(),
+				inputs.Template+".*.liquid",
+				cli.emailTemplateEditorHint,
+			); err != nil {
+				return err
+			}
+			if err != nil {
+				return fmt.Errorf("Failed to capture input from the editor: %w", err)
+			}
+
+			if !ruleEnabled.IsSet(cmd) {
+				inputs.Enabled = auth0.BoolValue(current.Enabled)
+			}
+
+			if err := ruleEnabled.AskBoolU(cmd, &inputs.Enabled, current.Enabled); err != nil {
+				return err
+			}
+
+			if inputs.From == "" {
+				inputs.From = current.GetFrom()
+			}
+
+			if inputs.Subject == "" {
+				inputs.Subject = current.GetSubject()
+			}
+
+			template := apiEmailTemplateFor(inputs.Template)
+			// Prepare email template payload for update. This will also be
+			// re-hydrated by the SDK, which we'll use below during
+			// display.
+			emailTemplate := &management.EmailTemplate{
+				Template: &template,
+				Body: &inputs.Body,
+				From: &inputs.From,
+				Subject: &inputs.Subject,
+				Enabled: &inputs.Enabled,
+			}
+
+			if inputs.ResultURL == "" {
+				emailTemplate.ResultURL = current.ResultURL
+			} else {
+				emailTemplate.ResultURL = &inputs.ResultURL
+			}
+
+			if inputs.ResultURLLifetime == 0 {
+				emailTemplate.URLLifetimeInSecoonds = current.URLLifetimeInSecoonds
+			} else {
+				emailTemplate.URLLifetimeInSecoonds = &inputs.ResultURLLifetime
+			}
+
+			if err = ansi.Waiting(func() error {
+				return cli.api.EmailTemplate.Update(template, emailTemplate)
+			}); err != nil {
+				return err
+			}
+
+			cli.renderer.EmailTemplateUpdate(emailTemplate)
+			return nil
+		},
+	}
+
+	emailTemplateBody.RegisterStringU(cmd, &inputs.Body, "")
+	emailTemplateFrom.RegisterStringU(cmd, &inputs.From, "")
+	emailTemplateSubject.RegisterStringU(cmd, &inputs.Subject, "")
+	emailTemplateEnabled.RegisterBoolU(cmd, &inputs.Enabled, true)
+	emailTemplateURL.RegisterStringU(cmd, &inputs.ResultURL, "")
+	emailTemplateLifetime.RegisterIntU(cmd, &inputs.ResultURLLifetime, 0)
+
+	return cmd
+}
+
+func (c *cli) emailTemplateEditorHint() {
+	c.renderer.Infof("%s once you close the editor, the email template will be saved. To cancel, CTRL+C.", ansi.Faint("Hint:"))
+}
+
+func (c *cli) emailTemplatePickerOptions() (pickerOptions, error) {
+	return emailTemplateOptions, nil
+}
+
+func apiEmailTemplateFor(v string) string {
```

Comment:
this is hard work. I'm wondering how we're going to manage breaking changes regarding of the hardcoded values here

Reply:
We could deprecate values and still accept the old one (and use the new one internally) and print a warning. And eventually stop accepting the old value.
There are a couple of ways we could go about the last step:
- Remove support for deprecated values in new majors
- Establish a policy for deprecations, where a deprecated value is guaranteed to be accepted for x time, and then can be removed
- ... (suggestions?)